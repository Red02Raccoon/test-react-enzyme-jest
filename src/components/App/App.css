.App {
  text-align: center;
}

.App-container {
    max-width: 980px;
    width: 100%;
    margin: 0 auto;

}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.App-block {
    border: 1px solid black;
    padding: 30px;
    margin-bottom: 30px;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
