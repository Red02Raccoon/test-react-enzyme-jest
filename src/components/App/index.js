import React, { Component } from "react";

import Form from "../Form";

import logo from "./logo.svg";
import "./App.css";

class App extends Component {
	state = {
		on: false,
		input: "",
		mainColor: "blue",
		lifeCycle: "",
	};
	handleStrings = str => {
		if (str === "Hello World") return true;
		return false;
	};
	componentDidMount() {
		this.setState({ lifeCycle: "componentDidMount" });
	}
	render() {
		const { mainColor, on, input, lifeCycle } = this.state;
		return (
			<div className="App">
				<header className="App-header">
					<img src={logo} className="App-logo" alt="logo" />
					<h1 className="App-title">Welcome to React</h1>
					<h3 className={mainColor}>Everyone is Welcome!</h3>
				</header>

				<div className="App-container">
					<Form />

					<div className="App-block">
						<p className="App-intro">Hello World</p>
						<p className="button-state">{on ? "Yes!" : "No!"}</p>
						<button onClick={() => this.setState({ on: true })}>Click</button>
						<h2>{input}</h2>
						<input onChange={e => this.setState({ input: e.currentTarget.value })} type="text" />
						<p className="lifeCycle">{lifeCycle}</p>
					</div>
				</div>
			</div>
		);
	}
}

export class Link extends Component {
	render() {
		return this.props.hide ? null : <a href={this.props.address}>Click</a>;
	}
}

export default App;
