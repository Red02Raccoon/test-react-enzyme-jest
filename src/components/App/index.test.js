import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import App, { Link } from "./";

describe("<App /> shallow rendering", () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<App />);
	});

	it("h1 contains correct text", () => {
		expect(wrapper.find("h1").text()).toBe("Welcome to React");
	});
	it("matches the snapshot", () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});
	it("updates className with new State", () => {
		expect(wrapper.find(".blue").length).toBe(1);
		expect(wrapper.find(".red").length).toBe(0);
		wrapper.setState({ mainColor: "red" });
		expect(wrapper.find(".blue").length).toBe(0);
		expect(wrapper.find(".red").length).toBe(1);
	});
	it("on button click changes p text", () => {
		const button = wrapper.find("button");
		expect(wrapper.find(".button-state").text()).toBe("No!");
		button.simulate("click");
		expect(wrapper.find(".button-state").text()).toBe("Yes!");
	});
	it("on input change, title changes text", () => {
		const mock = {
			value: "Tyler",
		};
		const input = wrapper.find("input");
		expect(wrapper.find("h2").text()).toBe("");

		input.simulate("change", { currentTarget: { value: mock.value } });
		expect(wrapper.find("h2").text()).toBe(mock.value);
	});
	it("calls componentDidMount, updates p tag text", () => {
		jest.spyOn(App.prototype, "componentDidMount");
		const wrapper = shallow(<App />);
		expect(App.prototype.componentDidMount.mock.calls.length).toBe(1);
		expect(wrapper.find(".lifeCycle").text()).toBe("componentDidMount");
	});
	it("handleStrings function returns correctly", () => {
		const trueReturn = wrapper.instance().handleStrings("Hello World");
		const falseReturn = wrapper.instance().handleStrings("");
		expect(trueReturn).toBe(true);
		expect(falseReturn).toBe(false);
	});
});

describe("<App /> mount rendering", () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mount(<App />);
	});
	it("h1 contains correct text", () => {
		expect(wrapper.find("h1").text()).toBe("Welcome to React");
	});
	it("matches the snapshot", () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	afterEach(() => {
		wrapper.unmount();
	});
});

describe("<Link />", () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<Link address="www.google.com" hide={false} />);
	});

	it("link component accepts address prop", () => {
		expect(wrapper.instance().props.address).toBe("www.google.com");
	});
	it("a tag node renders href correctly", () => {
		expect(wrapper.props().href).toBe("www.google.com");
	});
	it("returns null with true hide prop", () => {
		expect(wrapper.find("a").length).toBe(1);
		wrapper.setProps({ hide: true });
		expect(wrapper.get(0)).toBeNull();
	});
});
