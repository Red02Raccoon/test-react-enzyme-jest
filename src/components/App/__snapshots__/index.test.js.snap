// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
      <h3
        className="blue"
      >
        Everyone is Welcome!
      </h3>
    </header>
    <div
      className="App-container"
    >
      <Form>
        <form
          data-testid="addUserForm"
          onSubmit={[Function]}
        >
          <h2>
            Request Information
          </h2>
          <input
            data-testid="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
          <input
            data-testid="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          <input
            data-testid="number"
            onChange={[Function]}
            placeholder="Number"
            type="text"
            value=""
          />
          <div>
            <input
              checked={true}
              data-testid="checked"
              onChange={[Function]}
              onClick={[Function]}
              type="checkbox"
            />
            <p
              className="promotions"
              data-testid="promotionsP"
            >
              Receive Promotions
            </p>
          </div>
          <button
            data-testid="submitButton"
            type="submit"
          >
            Submit
          </button>
        </form>
      </Form>
      <div
        className="App-block"
      >
        <p
          className="App-intro"
        >
          Hello World
        </p>
        <p
          className="button-state"
        >
          No!
        </p>
        <button
          onClick={[Function]}
        >
          Click
        </button>
        <h2 />
        <input
          onChange={[Function]}
          type="text"
        />
        <p
          className="lifeCycle"
        >
          componentDidMount
        </p>
      </div>
    </div>
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h3
      className="blue"
    >
      Everyone is Welcome!
    </h3>
  </header>
  <div
    className="App-container"
  >
    <Form />
    <div
      className="App-block"
    >
      <p
        className="App-intro"
      >
        Hello World
      </p>
      <p
        className="button-state"
      >
        No!
      </p>
      <button
        onClick={[Function]}
      >
        Click
      </button>
      <h2 />
      <input
        onChange={[Function]}
        type="text"
      />
      <p
        className="lifeCycle"
      >
        componentDidMount
      </p>
    </div>
  </div>
</div>
`;
